esp-hal/src/uart.rs

```diff
@@ -117,6 +114,20 @@ impl embedded_io::Error for Error {
     }
 }
 
+// Define an enum for the clock source
+// (outside of `config` module in order not to "use" it an extra time)
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum ClockSource {
```

Comment:
Could you please add doc comments for this enum and for its variants?

Reply:
Done